extensions/stackdriver/edges/edge_reporter.h

```diff
@@ -0,0 +1,95 @@
+/* Copyright 2019 Istio Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <string>
+#include <vector>
+
+#include "extensions/common/context.h"
+#include "extensions/stackdriver/edges/edges.pb.h"
+#include "extensions/stackdriver/edges/mesh_edges_service_client.h"
+#include "google/protobuf/util/time_util.h"
+
+namespace Extensions {
+namespace Stackdriver {
+namespace Edges {
+
+#ifdef NULL_PLUGIN
+using Envoy::Extensions::Common::Wasm::Null::Plugin::Extensions::Stackdriver::
+    Edges::MeshEdgesServiceClient;
+#endif
+
+using Envoy::Extensions::Common::Wasm::Null::Plugin::getCurrentTimeNanoseconds;
+using google::cloud::meshtelemetry::v1alpha1::ReportTrafficAssertionsRequest;
+using google::cloud::meshtelemetry::v1alpha1::WorkloadInstance;
+using google::protobuf::util::TimeUtil;
+
+namespace {
```

Comment:
I think it is not recommended to use unnamed namespace in header file: https://google.github.io/styleguide/cppguide.html#Unnamed_Namespaces_and_Static_Variables. The way I can think of is to make it a lambda:
```
class EdgeReporter {
 private:
  TimestampFn now_ = []() {
       return TimeUtil::NanosecondsToTimestamp(getCurrentTimeNanoseconds());
   }
}
```
then just pass in nullptr in actual logic and pass in proto current time function in test. like:
```
EdgeReporter(node_info, client, nullptr)
```

Reply:
I added another ctor and pass the lambda in that way. let me know if that's ok.